### DISCUSSION TITLE
How can I read a signal unreactively from a `createRenderEffect`?

### DISCUSSION THREAD

**Discussion by danieltroger** [solidjs/solid#647] [Q&A] [answered]
So basically I have a signal (A) and through reactivity this happens: A -> B (memo) -> C (render effect). Now in C I want to know the value of A without passing it through B (I have no idea what I'm doing) but if I use A in C, C will run twice when A updates - once through B and once just standalone. I do need the value of A in C tho. So now I made a variable and added another render effect to A that sets the variable which I then read in C. But there has to be a better way? Can't you add like `signal.current_value` or something which gives the current value without reactivity?

**Comment by atk** [comment, marked as answer]
untrack(A) should do the trick: https://www.solidjs.com/docs/latest/api#untrack

**Reply by danieltroger** [reply]
Oh wow that's epic, thanks

**Reply by danieltroger** [reply]
It does exactly what I want, w00t w00t

**Comment by danieltroger** [comment]
@atk yo how can I write to a signal unreactively, is that possible?

**Reply by atk** [reply]
You want to update the value without notifying any subscriber, if I understand you correctly. Something that Solid supports, which comes 
pretty close to your requirement, is `useTransition` (see https://solidjs.com/docs/latest/api#usetransition), which allows you to wait for the writing to trigger the reactivity.

**Reply by danieltroger** [reply]
Thanks! I don't understand what I'm doing yet but I also found this https://www.solidjs.com/docs/latest/api#batch it might sound better? Idk what the difference is, I don't know what suspense is yet.

**Reply by atk** [reply]
`batch` basically does the same, but only for synchronous updates that require costy calculations. For example, if you update the classList, a data attribute containing the result of a costy calculation and a property, using batch will ensure that all those are done within the same render frame. `useTransition` does it for asynchronous updates.

**Reply by ryansolid** [reply]
Generally speaking you don't. It's up to the listener to decide whether they are listening. But the producer can't silently update. It can happen naturally if you set the same value like put it in an object and update a property, but generally there is no mechanism to say don't notify. If the intention is to group updates. `batch` is what you are looking for.

**Reply by danieltroger** [reply]
Thanks, I understand! So `batch` would have been the right thing. I was working on the filtering of categories where changing the filter triggered a slow refetch from bakend. I wanted to implement that if you check a parent category the subcategories' boxes get checked, without it making a request for every checked subcategory.

In the end I ended up decoupling checking the boxes and refetching anyways, so I didn't end up using it but I will use batch next time
